### ISSUE TITLE
Review feedback

### ISSUE THREAD

**Issue by yeekangc** [OpenLiberty/draft-guide-microprofile-reactive-messaging#61]
- This is an introductory guide.  Ensure that we focus on the basics to get started with MP RM.  The title and abstract should reflect that.  The focus should be clear throughout too.

- I will suggest talking about what is async and reactive first before MicroProfile and Kafka.  Reactive isn't defined or explained at the moment.  Should be covered.  Consider a flow like Intro > What is Async; What is Reactive > What is MP; What is Kafka > App.

- Concepts section: Great to explain key concepts first.  Then, we need to tie it to the app or what users will build and learn appropriately.

- Write code first and then configure the Kafka connector (and after that the server).  This is a more natural flow IMO especially for developers.

- Section titles: Consider calling out what in terms of MP RM one will build besides the service name.

- "Trying the application":  Should be "Testing the application" still.

FYI, @lauracowen @gkwan-ibm.

**Comment by yeekangc** [comment]
I wonder if Additional Prerequisites should come before everything else now to avoid breaking the flow.  This is likely a separate issue that needs design but we should consider.  Perhaps this can be in the RH column.

**Comment by ankitagrawa** [comment]
Feedback addressed.

**Comment by yeekangc** [comment]
- What is "it's propagation of change"?  We have a typo?

> Reactive programming is a programming paradigm that is designed around asynchronous data streams and it’s propagation of change

- What do we mean by "temporal decoupling" and "temporal coupling"?  Can we rephrase to make it easier to understand?

- Do we need to point users to the specification?  What do they need from the spec?

> See the MicroProfile Reactive Messaging Specification for more information.

- What is "system load average data"?  Can we explain this in simpler terms?  Also, the first picture shows "CPU status message".  How do they map to each other?

- We have 2 instances of "channel systemLoad" in the Connector picture.  What are they trying to convey?

- Can we rephrase the below in simpler terms?

> The average system load for a system is calculated by taking the sum of the count of runnable items queued to the available processors and the count of runnable items running on the available processors averaged over a period of time.

- Why do we need to use RxJava?  Can we explain that?

> The Flowable.interval() from rxJava is used to create a Publisher. 

- Explain the below in terms of dependencies that one needs to add?  And, why are they needed where still necessary?

> In order to develop the MicroProfile Reactive Messaging application using Maven, add the microprofile-reactive-messaging-api, kafka-clients, and rxjava dependencies to the pom.xml

- Can we explain in terms of inventory as the consumer of events and then it tracks the system and usage data?

> The inventory microservice stores the information about the CPU usage that runs on different systems.

- For the section on "Configuring the MicroProfile Reactive Messaging connectors", can we start by explaining why we need to do this?

- Can we explain what is a "consumer group"?  Also, the paragraph below is an awkward break from the one before especially if we are referencing the properties specified with "all these properties".

> The group.id defines a unique name for the consumer group. All these properties are required by the Apache Kafka Producer Configs and Apache Kafka Consumer Configs.

- Note that Dockerfiles are already provided?

- What is "use @GET ..."?  Should we provide a hotlink?  Should we give an example of a "hostId" too?

> You can use the @GET http://localhost:9085/inventory/system/{hostId} URL to see the CPU systemLoad property for one particular system.

- Review front matter and what is listed for What's Next.

**Comment by ankitagrawa** [comment]
> Do we need to point users to the specification? What do they need from the spec?
> 
> See the MicroProfile Reactive Messaging Specification for more information.
> 

If they want to look at something more than what's in the guide. For ex . different method signatures have different behaviour and Invocation rules. There are many such things. Also it's an review comment to add the spec link.

> What is "system load average data"? Can we explain this in simpler terms? Also, the first picture shows "CPU status message". How do they map to each other?

system load average data is one type of CPU status. That's how they are mapped.

> We have 2 instances of "channel systemLoad" in the Connector picture. What are they trying to convey?

That one of them pushes to Kafka and the other one receives from Kafka.

> Why do we need to use RxJava? Can we explain that?
> The Flowable.interval() from rxJava is used to create a Publisher.

So that the properties are sent every 15 seconds from the System. Also we need a publisher which is accomplished by using Flowable.interval().

> Explain the below in terms of dependencies that one needs to add? And, why are they needed where still necessary?
> In order to develop the MicroProfile Reactive Messaging application using Maven, add the microprofile-reactive-messaging-api, kafka-clients, and rxjava dependencies to the pom.xml

Didn't get your point. Are you asking us to define why are these dependencies required? I am assuming the basic guides like Maven and others talks about why the dependencies are required.

> Can we explain in terms of inventory as the consumer of events and then it tracks the system and usage data?
> The inventory microservice stores the information about the CPU usage that runs on different systems.

Aren't we doing that?  The title "Building the Consumer in the inventory microservice" suggests that. We can rephrase some of the sentences.

> For the section on "Configuring the MicroProfile Reactive Messaging connectors", can we start by explaining why we need to do this?

Just couple of lines into the section we have this sentence. Both the `system` and `inventory` microservices use connectors to connect certain channels to Apache Kafka and these connectors are configured by setting properties using MicroProfile Config. This is done by setting the required properties inside the `microprofile-config.properties` file included in each microservice. Isn't that sufficient?  Adding similar content in the beginning of the section could duplicate things.

> Can we explain what is a "consumer group"? Also, the paragraph below is an awkward break from the one before especially if we are referencing the properties specified with "all these properties".
> 
> The group.id defines a unique name for the consumer group. All these properties are required by the Apache Kafka Producer Configs and Apache Kafka Consumer Configs.

We can but Consumer group term is related more to Kafka. We have a guide on Kafka in the pipeline. The paragraph before doesn't have a group.id. so I mention that in this paragraph and explain that these properties are required for the config. Bcoz in an application you would have at least one producer and consumer.

> Note that Dockerfiles are already provided?
I used to have a line in the readme similar to "Most of the code is provided for use". But one of the feedback wanted me to remove that.

> What is "use @GET ..."? Should we provide a hotlink? Should we give an example of a "hostId" too?
> 
> You can use the @GET http://localhost:9085/inventory/system/{hostId} URL to see the CPU systemLoad property for one particular system.

I don't see it as a hotlink. It's a greyed text. We haven't mentioned hostId in previous guides.

> Review front matter and what is listed for What's Next.

Once we have other guides published we will replace what's next with that guide. Since this is the first one in the series.

@yeekangc @gkwan-ibm

**Comment by ankitagrawa** [comment]
addressed the feedback.

**Comment by yeekangc** [comment]
Assuming #73 and #75 will be satisfactorily addressed and the additional comments will be considered, I am closing this issue based on the current changes merged.

- Review and ensure that all hot spots are tagged accordingly particularly for first appearance of an annotation or a code reference.
- Consider "Configuring Kafka connectors" in place "Configuring MicroProfile Reactive Messaging connectors".
